datadog/api/aws_integration.py

```diff
@@ -0,0 +1,222 @@
+        defines a filter that Datadog uses when collecting metrics from EC2. \
+        Wildcards, such as ? (for single characters) and * (for multiple characters) \
+        can also be used. Only hosts that match one of the defined tags will be imported \
+        into Datadog. The rest will be ignored. Host matching a given tag can also be \
+        excluded by adding ! before the tag. e.x. \
+        env:production,instance-type:c1.*,!region:us-east-1 For more information \
+        on EC2 tagging, see the AWS tagging documentation.
+        :type filter_tags: list of strings
+
+        :param host_tags: Array of tags (in the form key:value) to add to all hosts and \
+        metrics reporting through this integration.
+        :type host_tags: list of strings
+
+        :param account_specific_namespace_rules: An object (in the form \
+        {"namespace1":true/false, "namespace2":true/false}) that enables \
+        or disables metric collection for specific AWS namespaces for this \
+        AWS account only. A list of namespaces can be found at the \
+        /v1/integration/aws/available_namespace_rules endpoint.
+        :type account_specific_namespace_rules: dictionary
+
+        :returns: Dictionary representing the API's JSON response
+
+        The following will depend on whether role delegation or access keys are being used.
+        If using role delegation, use the fields for role_name and account_id.
+        For access keys, use fields for access_key_id and secret_access_key.
+
+        Both the existing fields and new fields are required no matter what. i.e. If the config is \
+        account_id/role_name based, then `account_id`, `role_name`, `new_account_id`, and \
+        `new_role_name` are all required.
+
+        For access_key based accounts, `access_key_id`, `secret_access_key`, `new_access_key_id`, \
+        and `new_secret_access_key` are all required.
+
+        >>> account_id = "<EXISTING_AWS_ACCOUNT_ID>"
+        >>> role_name = "<EXISTING_AWS_ROLE_NAME>"
+        >>> access_key_id = "<EXISTING_AWS_ACCESS_KEY_ID>"
+        >>> secret_access_key = "<EXISTING_AWS_SECRET_ACCESS_KEY>"
+        >>> new_account_id = "<NEW_AWS_ACCOUNT_ID>"
+        >>> new_role_name = "<NEW_AWS_ROLE_NAME>"
+        >>> new_access_key_id = "<NEW_AWS_ACCESS_KEY_ID>"
+        >>> new_secret_access_key = "<NEW_AWS_SECRET_ACCESS_KEY_ID>"
+        >>> filter_tags = ["<KEY>:<VALUE>"]
+        >>> host_tags = ["<KEY>:<VALUE>"]
+        >>> account_specific_namespace_rules = {"namespace1":true/false, "namespace2":true/false}
+
+        >>> api.AwsIntegration.update(account_id=account_id, role_name=role_name, \
+        new_account_id=new_account_id, new_role_name=new_role_name, \
+        filter_tags=filter_tags,host_tags=host_tags,\
+        account_specific_namespace_rules=account_specific_namespace_rules)
+        """
+        params = {}
+        if body.get('account_id') and body.get('role_name'):
```

Comment:
If somebody wants to ever pass `None` as a value for e.g `role_name` for whatever reason, that wouldn't work.

As rule of thumb, it's safer to go with:
```python
if 'role_name' in body:
    params['role_name'] = body.pop('role_name')
```

Reply:
If they they were to try and pass in `role_name` as `None`, they would be met with the following error from the API:

`'errors': [u'AWS Account does not exist']`

That's to be expected. It's not currently possible to have `None` saved as the role_name so this will be the case. 

That said, I had thought that `.get` will return `None` if the `role_name` field isn't present, so wouldn't it essentially be the same as if they didn't submit a `role_name` at all?

Reply:
My point is that you either 1. implement a full logic in this method, or you 2. do the bare minimum and pass the data to the API and let it validate.
I think you're trying to do 2., which is I think the best solution in this case, but you're just preventing a (probably useless) case to be possible.

It's really a detail and I don't think it'll be an issue for any user, so feel free to ignore. :)

Reply:
Okay cool. Yeah we have fairly extensive validation on our end in the API, so that's kinda what I was going for here. Thanks @jd! I really appreciate the review 🙇!